shimmy/utils/meltingpot.py

```diff
@@ -0,0 +1,61 @@
+"""Shared utils for meltingpot."""
+
+from typing import Any, Mapping
+
+import dm_env
+from gymnasium import spaces
+import numpy as np
+import tree
+
+PLAYER_STR_FORMAT = 'player_{index}'
+_WORLD_PREFIX = 'WORLD.'
+
+
+def timestep_to_observations(timestep: dm_env.TimeStep) -> Mapping[str, Any]:
+  gym_observations = {}
+  for index, observation in enumerate(timestep.observation):
+    gym_observations[PLAYER_STR_FORMAT.format(index=index)] = {
+        key: value
+        for key, value in observation.items()
+        if _WORLD_PREFIX not in key
+    }
+  return gym_observations
+
+
+def remove_world_observations_from_space(
+    observation: spaces.Dict) -> spaces.Dict:
+  return spaces.Dict({
+      key: observation[key] for key in observation if _WORLD_PREFIX not in key
+  })
+
+
+def spec_to_space(spec: tree.Structure[dm_env.specs.Array]) -> spaces.Space:
```

Comment:
As jet mentioned, remove and replace with the existing dm spec to space

Reply:
So the existing spec to space is actually less general than this one, and is a bit less clear (imo) so I’ve actually replaced the existing one with this and adapted the functionality to work with both cases. Will commit the changes later today.